### PR TITLE
ci: add tox testenv for ruff

### PR DESCRIPTION
**Pull request by RogerHYang**
```shell
uvx --with tox-uv tox r -e ruff
```

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -139,3 +139,10 @@ commands_pre =
     --compile-bytecode
 commands =
   uv tool run arize-phoenix {posargs:serve}
+
+[testenv:ruff]
```
Comment (axiomofjoy) [inline tox.ini:143]:
```suggestion
[testenv:format]
```

Comment (RogerHYang) [reply tox.ini:143]: it's also got linting though

File: .github/workflows/python-CI.yml [hunk 2]
```diff
@@ -268,16 +258,9 @@ jobs:
...
-            pyproject.toml
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Install `ruff`
-        run: uv tool install ruff@0.6.8
-      - name: Check format
-        run: uv tool run ruff format --diff
-      - name: Check lint
-        run: uv tool run ruff check --no-fix
+      - name: Run `ruff`
+        run: uvx --with tox-uv tox run -e ruff
```
Comment (axiomofjoy) [inline .github/workflows/python-CI.yml:262]:
```suggestion
      - name: Run formatter
        run: uvx --with tox-uv tox run -e format
```

**Review by axiomofjoy — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (axiomofjoy) [reply tox.ini:143]: non-blocking

Comment (RogerHYang) [reply tox.ini:143]: e.g. [this](https://docs.astral.sh/ruff/rules/numpy2-deprecation/) would be lint, and not format

**Pull request merged**